Combine recent PRs and add tests for it all

Peter-Darton-i2 opened the pull request:
The existing orb is hard to test because:
1. the jobs are too inflexible (fixed by #49)
1. a bit broken (fixed in #46)
1. used orb-tools@11 to build it (fixed in #45)

This PR combines all of the above, plus #47, #48 and #50 and adds tests for all of these (which wasn't possible until it was all brought together).

Note: This should not be merged until after all of the above have been merged.

marboledacci commented:
Hi @Peter-Darton-i2, I have already merged all your previous PRs. I think for this is better if you merge main to your branch first, as this is a big PR and has some conflicts that you will understand better.
After that I can review it.
One think to remember, in the PR updating to orb-tools 12 I updated the name of one job to be test, this is because the repo  settings are requiring one check to be called test, so this has to be kept. It doesn't matter which of the new tests you added is it, but one have to be called test.

Peter-Darton-i2 commented:
@marboledacci Understood.
I have rebased this PR so it's much simpler to see what the _actual_ changes are.

I also changed which job is called "test" to the final `orb-tools/pack` job so that GitHub will only consider the build to have "passed" if _all_ these new tests pass.

Peter-Darton-i2 commented:
@marboledacci Thank you very much for the speedy resolution.
I was also going to ask "Can you make a new release?" ... but I see that you already have! 😀 

10/10 - much appreciated - you've been awesome, thank you.
